src/persist/src/indexed/cache.rs

```diff
@@ -138,6 +138,14 @@ impl<B: Blob> BlobCache<B> {
         Ok(val_len)
     }
 
+    /// Removes a batch from both [Blob] storage and the local cache.
+    pub fn delete_unsealed_batch(&mut self, key: &str) -> Result<(), Error> {
+        self.unsealed.lock()?.remove(key);
+        self.blob.lock()?.delete(key)?;
```

Comment:
Let's add blob_delete_ms/blob_delete_count/blob_delete_bytes metrics

Reply:
I added everything but the third - couldn't think of a way to do it without either plumbing the data size through somehow or doing a get before the delete neither of which seeemed particularly appealing.

Do you have any good ideas / otherwise how do you feel about leaving it as a todo?

Reply:
Ah bummer, yeah. This one is going to be important for reasoning about how much data we have outstanding, but i'll be in here again with some metrics stuff shortly, so I suppose it's okay to leave it as a TODO.

My idea is that we store size_bytes on the batch metadata now, but I think we probably want to add some validations to make sure that blob_write_bytes and size_bytes and this would all line up

Reply:
spitball: we could keep the size of each key in a separate hashtable as part of the BlobCache object, that way at delete time we know exactly how many bytes we get back and it could make it easier to assert global invariants over the size of everything.

It would require a `list` operation to hydrate the size info again on restart somehow though which maybe sucks